tartufo/scanner.py

```diff
@@ -247,13 +248,38 @@ def included_paths(self) -> List[Pattern]:
         """
         if self._included_paths is None:
             self.logger.info("Initializing included paths")
-            patterns = list(self.global_options.include_path_patterns or ())
-            self._included_paths = (
-                config.compile_path_rules(set(patterns)) if patterns else []
-            )
-            self.logger.debug(
-                "Included paths was initialized as: %s", self._included_paths
-            )
+            patterns = list(self.global_options.include_path_patterns or ())  # type: ignore
+            try:
+                patterns = [
+                    pattern["path-pattern"]
+                    for pattern in cast(List[Dict[str, str]], patterns)
+                ]
+                self._included_paths = (
+                    config.compile_path_rules(set(cast(List[str], patterns)))
+                    if patterns
+                    else []
+                )
+                return self._included_paths
+            except TypeError:
+                warnings.warn(
+                    "Old format of --include-path-patterns option and config file setup include-path-patterns "
+                    "= ['inclusion pattern'] has been deprecated and will be removed in a future version. "
+                    "Make sure all the inclusions are set up using new pattern i.e. include-path-patterns = "
+                    "[path-pattern='inclusion pattern',reason='reason for inclusion'] in the config file",
+                    DeprecationWarning,
+                )
+            try:
+                self._included_paths = (
+                    config.compile_path_rules(set(cast(List[str], patterns)))
+                    if patterns
+                    else []
+                )
+                return self._included_paths
+            except TypeError as exc:
+                raise types.ConfigException(
+                    f"Combination of old and new format of include-path-patterns will not be supported.\n{exc}"
+                )
+        self.logger.debug("Included paths was initialized as: %s", self._included_paths)
```

Comment:
```suggestion
            self.logger.debug("Included paths was initialized as: %s", self._included_paths)
```
We don't need this to log every single time `included_paths` is accessed 😄